index.bs

```diff
@@ -641,6 +641,93 @@ To <dfn>obtain a report</dfn> given an [=attribution source=] |source| and an [=
     :: |source|'s [=attribution source/source identifier=].
 1. Return |report|.
 
+# Report delivery # {#report-delivery}
+
+The user agent MUST periodically gather [=attribution reports=] from its [=attribution report cache=] and run [=queue a report for delivery=] on each report.
+
+To <dfn>queue a report for delivery</dfn> given an [=attribution report=] |report| run the following steps:
+
+1. If |report|'s [=attribution report/report time=] is less than the current time, add an [=implementation-defined=] random amount to report time.
+    
+    Note: On startup, it is possible the user agent will need to send many reports whose report times passed while the browser was 
+     closed. Adding random delay prevents temporal joining of reports from different [=attribution source/source origin=]'s.
+1. Wait until |report|'s [=attribution report/report time=] is the current time.
+1. Optionally, wait a further implementation-defined length of time.
+    
+    Note: This is intended to allow user agents to optimize device resource usage.
+1. Run [=attempt to deliver a report=] with |report|.
+
+<h3 id="serialize-report-body">Serialize attribution report body</h3>
+
+To <dfn>serialize an [=attribution report=]</dfn> |report|:
+
+1. Let |data| be a map of the following key/value pairs:
+
+    : `source_type`
+    :: report's [=attribution report/source type=]
+    : `source_event_id`
+    :: |report|'s [=attribution report/event id=]
+    : `trigger_data`
+    :: |report|'s [=attribution report/trigger data=]
+
+1. Return the [=byte sequence=] resulting from executing [=serialize an infra value to JSON bytes=] on |data|.
+
+<h3 id="get-report-url">Get report request URL</h3>
+
+To <dfn>generate a report URL</dfn> for a given [=attribution report=] |report|:
+
+1. Let |serializedOrigin| be the result of executing [=serialization of an origin=] with |report|'s [=attribution report/reporting endpoint=].
+1. Let |urlComponents| be the list «|serializedOrigin|, ".well-known/attribution-reporting/report-attribution"».
+1. Let |urlString| be the result of executing [=string/concatenate=] on |urlComponents|.
+1. Return the result of running the [=URL parser=] on |urlString|.
+
+<h3 id="issue-report-request">Issuing a report request</h3>
+
+This algorithm constructs a [=request=] and attempts to deliver it to
+|report|'s [=attribution report/reporting endpoint=].
+
+To <dfn>attempt to deliver a report</dfn> given an [=attribution report=] |report| run the following steps:
+
+1. Let |body| be the result of executing [=serialize an attribution report=] on |report|.
+1. Let |url| be the result of executing [=generate a report url=] on |report|.
+1. Let |referrer| the result of running the [=URL parser=] on |report|'s [=attribution report/attribution destination=].
+1. Let |request| be a new [=request=] with the following properties [[FETCH]]:
```

Comment:
`[[FETCH]]` is being render as `[FETCH]` in the preview. Is that intentional?

Reply:
I don't think this ref was necessary, removed.

Reply:
Looks like it's still there?